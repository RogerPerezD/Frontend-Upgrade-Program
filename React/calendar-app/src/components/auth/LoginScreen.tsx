import React from 'react';
import '../../css/login.css';
import { useForm } from '../../hooks/useForm';
import { FormEvent } from 'react';
import { useDispatch } from 'react-redux';
import { startLogin, startRegister } from '../../actions/auth';
import Swal from 'sweetalert2';

type LoginForm = {
    lEmail: string,
    lPassword: string
}

type RegisterForm = {
    rEmail: string,
    rPassword: string,
    rPasswordVerified: string,
    rName: string,
}

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [ formLoginValues, handleLoginInputChange ]= useForm<LoginForm>({
        lEmail: 'rogelio@mail.com',
        lPassword: '1234567'
    });

    const [ formRegisterValues, handleRegisterInputChange ]= useForm<RegisterForm>({
        rEmail: 'raul@mail.com',
        rPassword: '123456',
        rPasswordVerified: '123456',
        rName: 'Raul'
    });

    const { lEmail, lPassword } = formLoginValues;

    const { rName, rEmail, rPassword, rPasswordVerified} = formRegisterValues;

    const handleLogin = ( e: FormEvent)=>{
        e.preventDefault();
        console.log(lEmail, lPassword,'valores login');
        dispatch( startLogin(lEmail, lPassword) );
    }

    const handleRegister = ( e: FormEvent)=>{
        e.preventDefault();
        if (rPassword !== rPasswordVerified) {
            return Swal.fire('Error', 'Passwords must match','error')
        }
        dispatch( startRegister(rEmail, rPassword, rName) );
    }
    return (
        <div className="container login-container">
            <div className="row">
                <div className="col-md-6 login-form-1">
                    <h3>Ingreso</h3>
                    <form onSubmit={ handleLogin }>
                        <div className="form-group">
                            <input 
                                type="text"
                                className="form-control"
                                placeholder="Correo"
                                name='lEmail'
                                value={lEmail}
                                onChange= {handleLoginInputChange}
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contraseña"
                                name='lPassword'
                                value={lPassword}
                                onChange= {handleLoginInputChange}
                            />
                        </div>
                        <div className="form-group">
                            <input 
                                type="submit"
                                className="btnSubmit"
                                value="Login" 
                            />
                        </div>
                    </form>
                </div>

                <div className="col-md-6 login-form-2">
                    <h3>Registro</h3>
                    <form onSubmit={ handleRegister }>
                        <div className="form-group">
                            <input
                                type="text"
                                className="form-control"
                                placeholder="Nombre"
                                name='rName'
                                value={rName}
                                onChange= { handleRegisterInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="email"
                                className="form-control"
                                placeholder="Correo"
                                name='rEmail'
                                value={rEmail}
                                onChange= { handleRegisterInputChange }
                            />
                        </div>
                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Contraseña"
                                name='rPassword' 
                                value={rPassword}
                                onChange= { handleRegisterInputChange }
                            />
                        </div>

                        <div className="form-group">
                            <input
                                type="password"
                                className="form-control"
                                placeholder="Repita la contraseña"
                                name='rPasswordVerified'
                                value={rPasswordVerified}
                                onChange= { handleRegisterInputChange } 
                            />
                        </div>

                        <div className="form-group">
                            <input 
                                type="submit" 
                                className="btnSubmit" 
                                value="Crear cuenta" />
                        </div>
                    </form>
                </div>
            </div>
        </div>
    )
}