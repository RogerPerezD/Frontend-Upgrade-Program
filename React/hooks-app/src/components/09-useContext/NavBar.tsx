import React from 'react';
import { Link } from "react-router-dom";

export const NavBar = () => {
  return (
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <Link className='navbar-brand' to="/">useContext</Link>

                <div className="collapse navbar-collapse" id="navbarNav">
                <ul className="navbar-nav">
                    <Link className="nav-link active" to="/">Home</Link>
                    <Link className="nav-link" to="about">About</Link>
                    <Link className="nav-link" to="login">Login</Link>
                </ul>
                </div>
            </div>
        </nav>
  );
};
