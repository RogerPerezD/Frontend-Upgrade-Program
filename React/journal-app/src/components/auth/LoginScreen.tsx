import { Link } from "react-router-dom";
import { useForm } from '../../hooks/useForm';
import { FormEvent } from 'react';
import { useDispatch } from "react-redux";
import { login, startLoginEmailPassword, startGoogleLogin } from '../../actions/auth';
// import { AppDispatch } from '../../store/store';

export const LoginScreen = () => {

    const dispatch = useDispatch();

    const [ formValues, handleInputChange ] = useForm({
        email: 'roger@mail.com',
        password: '1234'
    });


    const handleLogin = ( e: FormEvent) =>{
        e.preventDefault();
        console.log(email, password);
        // dispatch( login( '543bgds', 'Roger'));
        dispatch(startLoginEmailPassword( '543bgds', 'Roger'));
    }

    const handleGoogleLogin = () =>{
        dispatch( startGoogleLogin() );
    }

    const { email, password} = formValues;
    return (
        <>
            <h3 className="auth__title"> Login </h3>
            <form onSubmit={ handleLogin }>
                <input 
                    type="email" 
                    name="email" 
                    placeholder="Email" 
                    className="auth__input" 
                    autoComplete="off"
                    value={ email }
                    onChange={ handleInputChange}                    
                />
                <input 
                    type="password" 
                    name="password" 
                    placeholder="Password" 
                    className="auth__input" 
                    autoComplete="off"
                    value={ password }
                    onChange={ handleInputChange}
                />
                <button type="submit" className="btn btn-primary btn-block">
                    Login
                </button>

                <div className="auth__social-networks">
                    <p>Login with social networks</p>
                    <div 
                        className="google-btn"
                        onClick={ handleGoogleLogin }
                    >
                        <div className="google-icon-wrapper">
                            <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                        </div>
                        <p className="btn-text">
                            <b>Sign in with google</b>
                        </p>
                    </div>
                </div>

                <Link to="/auth/register" className="link">
                    Create New Account
                </Link>
            </form>
        </>
    );
};
