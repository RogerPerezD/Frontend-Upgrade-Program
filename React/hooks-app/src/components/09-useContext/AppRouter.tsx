import React from 'react';
import {
    BrowserRouter as Router,
    Routes,
    Route
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { LoginScreen } from './LoginScreen';
import { HomeScreen } from './HomeScreen';
import { NavBar } from './NavBar';
import { NotFoundScreen } from './NotFoundScreen';

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <div className="container">
          <Routes>
            <Route path="/"  element={ <HomeScreen />}/>
            <Route path="/about"  element={ <AboutScreen />}/>
            <Route path="/login"  element={ <LoginScreen />}/>
            <Route path="*"  element={ <NotFoundScreen />}/>
          </Routes>
        </div>
      </div>
    </Router>
  );
};
