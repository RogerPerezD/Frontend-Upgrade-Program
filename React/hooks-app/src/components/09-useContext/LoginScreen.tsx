import React, { useContext } from 'react';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

  const { setUser } = useContext(UserContext);

    const user= {
    id: 123,
    name: 'Roger',
    email: 'roger@mail.com'
  }

  return (
    <div>
      <h1>Login Screen</h1>
      <button 
      className='btn btn-primary'
      onClick={ ()=> { setUser( user )}}
      >
        Login
      </button>
      <hr />
    </div>
  );
};
