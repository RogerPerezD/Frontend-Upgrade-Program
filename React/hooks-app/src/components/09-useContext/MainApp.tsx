import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContext';

export interface User{
  id: number;
  name: string;
  email: string;
}

export const MainApp = () => {

  const [user, setUser] = useState<User>({} as User);


  return (
      <UserContext.Provider value={ {
        user,
        setUser
      } }>
        <AppRouter/>
      </UserContext.Provider>
      
  );
};
