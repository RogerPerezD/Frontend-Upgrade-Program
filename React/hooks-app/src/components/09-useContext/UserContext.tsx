import { createContext } from "react";
import { User } from "./MainApp";

type UserContexType = {
    user: User,
    setUser: React.Dispatch<React.SetStateAction<User>>
}


export const UserContext = createContext<UserContexType>({} as UserContexType);