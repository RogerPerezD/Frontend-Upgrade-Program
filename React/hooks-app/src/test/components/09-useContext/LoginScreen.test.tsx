import { mount } from "enzyme";
import { UserContext } from '../../../components/09-useContext/UserContext';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';

describe('Test in <LoginScreen/>', () => {
    

    const userDefault = {
        id: 1,
        name: 'Roger',
        email: 'roger@mail.com',
    }
    const setUser = jest.fn();
    const wrapper = mount( 
    <UserContext.Provider value={ { user: userDefault, setUser} }>  
        <LoginScreen/>
    </UserContext.Provider> );

    test('should show the component successfully', () => {
        
        expect( wrapper ).toMatchSnapshot();
    });

    test('should execute setUser whit the argument expected', () => {
        const user= {
            id: 123,
            name: 'Roger',
            email: 'roger@mail.com'
        };

        const button = wrapper.find('button');
        button.simulate('click');

        expect( setUser ).toBeCalledWith( user );
    });
    
    
});
