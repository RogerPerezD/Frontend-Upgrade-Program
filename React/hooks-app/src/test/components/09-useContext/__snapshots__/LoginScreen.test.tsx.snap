// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LoginScreen/> should show the component successfully 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
  <hr />
</div>
`;
